e2e/scripts/mrt-target-manager.js

```diff
@@ -0,0 +1,322 @@
+        return new Promise((resolve) => setTimeout(resolve, ms))
+    }
+}
+
+async function main() {
+    const program = new Command()
+
+    program
+        .description('Acquire and manage MRT environments with optimistic locking')
+        .option('--max-retries <number>', 'Maximum retry attempts', '3')
+        .option('--retry-delay <ms>', 'Delay between retries in milliseconds', '10000')
+
+    program
+        .command('status')
+        .description('Show pool status')
+        .action(async () => {
+            /**
+             * roleArn: [ARN - Amazon Resource Name] unique identifier for the 'Role' resource
+             * that the currently authenticated AWS user assumes to get permissions defined by policies attached to the role.
+             *
+             * roleSessionName: Arbitrary identifier used to point out which session did certain actions originate from.
+             * Typically used in logs [AWS Cloudwatch logs] like:
+             * - [GithubActions-E2E-CI] Created new resource pwa-kit-ci/demo.json in S3.
+             * or
+             * - [LocalDev] Downloaded resource pwa-kit-ci/demo.json from S3.
+             */
+            const mrtTargetManager = new MRTTargetManager({
+                bucket: process.env.AWS_S3_BUCKET,
+                poolDataFileKey: process.env.AWS_S3_POOL_DATA_FILE_KEY,
+                roleArn: process.env.AWS_ROLE_ARN,
+                region: process.env.AWS_REGION,
+                roleSessionName: process.env.CI ? 'GithubActions-E2E-CI' : 'LocalDev'
+            })
+
+            await mrtTargetManager.initialize()
+
+            try {
+                const status = await mrtTargetManager.getPoolStatus()
+
+                console.log('Pool status:', JSON.stringify(status, null, 2))
+            } catch (error) {
+                console.error('❌ Error:', error.message)
+                process.exit(1)
+            }
+        })
+
+    program
+        .command('acquire')
+        .description('Acquire an MRT environment')
+        .option('--pr-number <prNumber>', 'PR number')
+        .option('--branch <branch>', 'Branch name')
+        .option('--run-id <runId>', 'Run ID')
+        .action(async ({prNumber, branch, runId}) => {
+            const globalOpts = program.opts()
+
+            const mrtTargetManager = new MRTTargetManager({
+                bucket: process.env.AWS_S3_BUCKET,
+                poolDataFileKey: process.env.AWS_S3_POOL_DATA_FILE_KEY,
+                roleArn: process.env.AWS_ROLE_ARN,
+                region: process.env.AWS_REGION,
+                prNumber,
+                branch,
+                runId,
+                maxRetries: parseInt(globalOpts.maxRetries),
+                retryDelay: parseInt(globalOpts.retryDelay),
+                roleSessionName: process.env.CI ? 'GithubActions-E2E-CI' : 'LocalDev'
+            })
+
+            await mrtTargetManager.initialize()
+
+            await fs.ensureFile(MRT_TARGET_DETAILS_FILE)
```

Comment:
fs.ensureFile() creates an empty file, but fs.writeJson() will create the file anyway.

Reply:
So each workflow run is a new server instance like a blank computer.
So the parent directory won't always exist in which case directly calling `fs.writeJson()` will fail.
So `fs.ensureFile()` the directory structure and file exists before calling `fs.writeJson()`